src/mscorlib/src/System/TimeZoneInfo.cs

```diff
@@ -1934,9 +1934,36 @@ private class OffsetAndRule {
 
 
 #if PLATFORM_UNIX
-        static public TimeZoneInfo FindSystemTimeZoneById(string id) {
-            // UNIXTODO
-            throw new NotImplementedException();
+        static public TimeZoneInfo FindSystemTimeZoneById(string id)
+        {
+            if (string.IsNullOrEmpty(id))
+            {
+                throw new ArgumentNullException("id");
+            }
+
+            // Special case for Utc since there is code that expects only 1 instance of the UTC time zone info to be created.
+            // see CachedData.GetCorrespondingKind()
+            if (String.Compare(id, c_utcId, StringComparison.OrdinalIgnoreCase) == 0)
```

Comment:
String.Equals can take StringCOmparison parameter too. so yes I agree should use Equals here